[tplinksmarthome] Add new device: EP25

joshproehl opened the pull request:
The Kasa EP25 appears to be a version of the KP125 that lacks the LED. Because of this, and per the comments in #16027, this simply copies the KP125, without the LED channel.

This is my attempt at contributing to this project, so I'm sure I've missed something. I haven't run any testing or build steps, all I've done is use the code that I saw in PR #1171 (Adding the KP125) as a template and add the EP25 device. I'm hoping this at least provides a base for someone more familiar with the project to do a quick review and test so we can get the device added! As mentioned in #16027, the EP25 appears to behave properly when added manually as a KP125 device, but that is the only testing that I've done, I have *not* figured out how to take this PR into my live OpenHAB environment to test it.

If accepted, this closes #16027

lsiepel commented on bundles/org.openhab.binding.tplinksmarthome/README.md:
```diff
@@ -385,13 +391,13 @@ All devices support some of the following channels:
 | Channel Type ID     | Item Type                | Description                                    | Thing types supporting this channel                                                                                                                    |
 |---------------------|--------------------------|------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------|
-| switch              | Switch                   | Power the device on or off.                    | EP10, EP40, HS100, HS103, HS105, HS107, HS110, HS200, HS210, HS300, KP100, KP105, KP115, KP200, KP303, KP400, KP401, KS230, RE270K, RE370K             |
+| switch              | Switch                   | Power the device on or off.                    | EP10, EP25, EP40, HS100, HS103, HS105, HS107, HS110, HS200, HS210, HS300, KP100, KP105, KP115, KP200, KP303, KP400, KP401, KS230, RE270K, RE370K             |
```
```suggestion
| switch              | Switch                   | Power the device on or off.                    | EP10, EP25, EP40, HS100, HS103, HS105, HS107, HS110, HS200, HS210, HS300, KP100, KP105, KP115, KP200, KP303, KP400, KP401, KS230, RE270K, RE370K       |
```

joshproehl replied:
Added this change via a force-pushed change to the original commit. Table alignment in README.md looks good in raw mode when viewing the current commit for the PR.

lsiepel commented on bundles/org.openhab.binding.tplinksmarthome/README.md:
```diff
@@ -385,13 +391,13 @@ All devices support some of the following channels:
 | color               | Color                    | Set the color of the light.                    | KB130, KL125, KL130, KL135, KL400, KL430, LB130, LB230                                                                                                 |
-| power               | Number:Power             | Actual energy usage in Watt.                   | HS110, HS300, KLxxx, KP115, KP125, LBxxx,                                                                                                              |
-| eneryUsage          | Number:Energy            | Energy Usage in kWh.                           | HS110, HS300, KP115, KP125                                                                                                                             |
+| power               | Number:Power             | Actual energy usage in Watt.                   | EP25, HS110, HS300, KLxxx, KP115, KP125, LBxxx,                                                                                                              |
+| eneryUsage          | Number:Energy            | Energy Usage in kWh.                           | EP25, HS110, HS300, KP115, KP125                                                                                                                             |
```
```suggestion
| power               | Number:Power             | Actual energy usage in Watt.                   | EP25, HS110, HS300, KLxxx, KP115, KP125, LBxxx,                                                                                                        |
| eneryUsage          | Number:Energy            | Energy Usage in kWh.                           | EP25, HS110, HS300, KP115, KP125                                                                                                                       |
```

Removed some excess spaces, somehow this suggestion does not show it :-/ (it is about aligning the table again)

joshproehl replied:
Added this change via a force-pushed change to the original commit. Table alignment in README.md looks good in raw mode when viewing the current commit for the PR.

lsiepel reviewed (changes requested):
Thanks, Left two minor comments. Kind of funny that you managed to get this PR this far without even building the code. 
Not sure wich IDE you use, but if you got this far, the rest should not be that difficult. Let me know what steps you are missing. Once the build is completed, it will generate a JAR file. 
After you uninstall the current binding in you openHAB setup, you can drop this JAR into your addons folder and your custom build binding verison should be started and is ready to test.

joshproehl commented:
> Thanks, Left two minor comments.

I'll go in and try and get the table formatted. I'm struggling a little because my NeoVIM setup's Markdown linter wants to make some rather sweeping changes to the README file, but I can get that squared away. :-)

>  Kind of funny that you managed to get this PR this far without even building the code.

Duplicating a previous PR that does exactly the same thing can get you a long way. 

> After you uninstall the current binding in you openHAB setup, you can drop this JAR into your addons folder and your custom build binding verison should be started and is ready to test.

So I should remove the *all* the addons and run a complete build of the entire addons repo? Seems like perhaps I just need to spin up a 2nd OpenHAB instance, as that would be much easier. (I was hoping I could bump a version number and just build the tplinksmarthome portion of the bundle and have that take precedence...)
If I spin up an entire OpenHAB instance can I just build and deploy the tplinksmarthome portion of the bundle? (using `mvn clean install -pl :org.openhab.binding.astro` per the README.)

Other than running the standard integration tests, and confirming that my built bundle can recognize EP25's automatically and add/control them, are there any further tests I should be doing?

lsiepel commented:
> should remove the _all_ the addons and run a complete build of the entire addons repo? Seems like perhaps I just need to spin up a 2nd Ope



> r than running the standard integration tests, and confirming that my built bundle can recognize EP25's automatically and add/control them, are there any further tests I should be doing?

No need to build everything, just the one binding would be enough (like in the example you got from the docs). That will generate one jar that only holds that specific binding. No need for another openHAB instance. Uninstalling the binding will have no effect on your things/channels/items etc.

joshproehl commented:
Per feedback I have fixed the whitespace issue in the README.md table, and have added the required sign-off text to the commit message. I have also pushed an additional fix to include the correct power channels, and have the name match the translations file. (I think I copied the wrong XML somehow...)

`mvn spotless:check` seems to pass fine.

I have built the modified tplinksmarthome binding using `mvn clean install -pl :org.openhab.binding.tplinksmarthome` and replaced my stock version of the binding with this snapshot version on my live OpenHAB instance. 
So far it seems to be working. An EP25 was detected and automatically added to the inbox, and can be added and controlled, as well as show the correct wattage in the power channel as confirmed by checking the Kasa app.

If there are further checks I can do please let me know!

lsiepel reviewed (approved):
Thanks for this contribution.
LGTM
